include/deal.II/differentiation/sd/symengine_scalar_operations.h

```diff
@@ -146,8 +146,317 @@ namespace Differentiation
        */
       bool
       is_valid_substitution_symbol(const SymEngine::Basic &entry);
+
+      /**
+       * A convenience function to set the @p value associated with
+       * the @p symbol in the @p substitution_map.
+       *
+       * Using this function ensures that the @p symbol is one that is
+       * valid specifically for the purpose of symbolic substitution.
+       * It must therefore represent a symbol or symbolic derivative,
+       * otherwise an error will be thrown.
+       */
+      void
+      set_value_in_symbol_map(
+        types::substitution_map &                     substitution_map,
+        const SymEngine::RCP<const SymEngine::Basic> &symbol,
+        const SymEngine::RCP<const SymEngine::Basic> &value);
     } // namespace internal
 
+    /**
+     * Return a symbolic map that has the entry key given by @p symbol.
```

Comment:
```suggestion
     * Return a symbolic map that has a single entry with the key given by @p symbol.
```
What is the value?

Reply:
One cannot say - SymbolicType could be an `Expression` or a `Tensor<2,dim,Expression>` or something else. It takes on the default value of the called `add_to_symbol_map()` function.

Reply:
So is it a default-constructed object? That is a term I understand :-)

Reply:
Right now, this is the terminal function that's called:
```
    template <bool ignore_invalid_symbols, typename ValueType>
    void
    add_to_symbol_map(types::substitution_map &symbol_map,
                      const Expression &       symbol)
    {
      // Call the above function
      add_to_substitution_map<ignore_invalid_symbols>(
        symbol_map,
        symbol,
        dealii::internal::NumberType<ValueType>::value(0.0));
    }
```
So, sort of...